api/v1alpha1/openstackimage_types.go

```diff
@@ -55,6 +37,12 @@ type OpenStackImageSpec struct {
 	// and iso.
 	DiskFormat string `json:"diskFormat,omitempty"`
 
+	// ID is the OpenStack UUID of the resource. If left empty, the
+	// controller will create a new resource and populate this field. If
+	// manually populated, the controller will adopt the corresponding
+	// resource.
+	ID string `json:"id,omitempty"`
+
 	// MinDisk is the amount of disk space in GB that is required to boot
 	// the image.
 	MinDisk int `json:"minDisk,omitempty"`
```

Comment:
Should we update this to string?

Reply:
since it's optional, it should be `*int`

Reply:
I'm hesitant about changing this, since it's optional in OpenStack Image v2 API (for what I understand from https://docs.openstack.org/api-ref/image/v2/#create-image) but it is not optional in Gophercloud. IMHO we should stick to Gophercloud as much as possible.

Reply:
It’s optional in Gophercloud, in the sense that “zero” is interpreted as “no value” and it’s not sent to the API (because of the ‘omitempty’ tag): https://github.com/gophercloud/gophercloud/blob/fec8a6f991b57556fbc5f6650b312d70de95e8e8/openstack/imageservice/v2/images/requests.go#L178

Gophercloud has settled on a different way of treating optional values; in orc I think we want to try to be consistent and use pointers wherever there is an optional value. If we realise that it’s the wrong approach, we will change it!

In general I think we should stick with the OpenStack API rather than Gophercloud. In my view, Goohercloud is an implementation detail here.

Reply:
Otherwise, we can change our [internal convention](https://github.com/gophercloud/openstack-resource-controller/blob/main/CONTRIBUTING.md#how-to-write-a-new-crd) and say that optional fields are pointers _when their zero value is significant_. Which would essentially let us use bare values almost everywhere, except in booleans and in some, but not all, strings and numbers. What do you think?